align_system/algorithms/oracle_adm.py

```diff
@@ -0,0 +1,321 @@
+                    case 'Internal':
+                        return 'Pain Medications'
+                    case _:
+                        raise ValueError(f"Unknown injury type: {injury_name}")
+
+                return None
+
+            # Determine what treatment to apply, narrow down applicable characters
+            if 'treatment' not in action_to_take.parameters:
+                # Can only apply treatments if supplies are available
+                poss_treatments = [s.type for s in scenario_state.supplies if s.quantity > 0]
+                ta3_server_treatments = [
+                    'Burn Dressing', 'Decompression Needle', 'Hemostatic gauze', 'Nasopharyngeal airway',
+                    'Pressure bandage', 'Splint', 'Tourniquet', 'Pain Medications'
+                ]
+                poss_treatments = [t for t in poss_treatments if t in ta3_server_treatments]
+
+                # Limit possible characters to those with injuries with specified location and supplies available
+                would_be_used = set()
+                if 'location' in action_to_take.parameters:
+                    tmp = list()
+                    for c in poss_chars:
+                        for injury in c.injuries:
+                            if injury.location == action_to_take.parameters["location"]:
+                                potential_treatment = _get_treatment(poss_treatments, injury.name, injury.location)
+                                if potential_treatment is not None:
+                                    tmp.append(c)
+                                    would_be_used.add(potential_treatment)
+                                    break
+                    poss_chars = tmp
+                else:  # Location not determined yet, could pick any treatment based on given supplies and injuries
+                    tmp = list()
+                    for c in poss_chars:
+                        for injury in c.injuries:
+                            potential_treatment = _get_treatment(poss_treatments, injury.name, injury.location)
+                            if potential_treatment is not None:
+                                tmp.append(c)
+                                would_be_used.add(potential_treatment)
+                                break
+                    poss_chars = tmp
+
+                if not would_be_used:
+                    log.warn("No treatments left, defaulting to random treatment")
+                    action_to_take.parameters['treatment'] = random.choice(poss_treatments)
+                else:
+                    action_to_take.parameters['treatment'] = random.choice(list(would_be_used))
+                log.info(f"Selected treatment: {action_to_take.parameters['treatment']}")
+
+            if 'location' not in action_to_take.parameters:
+                potential_locations = set()
+
+                # Limit possible characters to those with injuries that make sense for chosen treatment
```

Comment:
I could be missing some potential condition, but would it simplify (or reduce) the code if when the action to take is an APPLY_TREATMENT action, pick a random character with an untreated injury first, and then pick one of the injuries and then choose the treatment and location to treat said injury (so more top-down rather than bottom-up?)

Reply:
I guess that would work. My thought process was that we may not have the ability to treat the injury that ends up getting selected. However, I guess we could just try again if that was the case.

Reply:
Thanks Emily, I think it's a fair bit easier to follow now (at least to my eyes) and seems a fair bit shorter as well.